core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -318,73 +338,103 @@ private void tryToMakeAdmin(User u) {
     }
 
     /**
-     * @return
-     *      null if failed. The browser is already redirected to retry by the time this method returns.
-     *      a valid {@link User} object if the user creation was successful.
+     * @return null if failed. The browser is already redirected to retry by the time this method returns. a valid
+     * {@link User} object if the user creation was successful.
      */
     private User createAccount(StaplerRequest req, StaplerResponse rsp, boolean selfRegistration, String formView) throws ServletException, IOException {
+        SignupInfo si = validateAccountCreationForm(req, selfRegistration);
+
+        if (si.errorMessage != null) {
+            // failed. ask the user to try again.
+            req.getView(this, formView).forward(req, rsp);
+            return null;
+        }
+
+        return createAccount(si);
+    }
+
+    /**
+     * @param req              the request to process
+     * @param selfRegistration whether this account creation was issued by the user themselves
```

Comment:
The call passes `false`, probably because there's no captcha in the setup wizard, but still weird.

Reply:
You're right. I just copied this parameter from the original method.. Turns out it value is either statically false or the class field `enableCaptcha`. I've renamed it to `validateCaptcha` in both signatures, and also completed the JavaDoc for the original method while I was at it.